lang/src/test/kotlin/org/partiql/lang/compiler/PartiQLCompilerPipelineSmokeTests.kt

```diff
@@ -0,0 +1,193 @@
+package org.partiql.lang.compiler
+
+import com.amazon.ionelement.api.ionInt
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import org.partiql.lang.ION
+import org.partiql.lang.ast.SourceLocationMeta
+import org.partiql.lang.domains.PartiqlPhysical
+import org.partiql.lang.errors.PartiQLException
+import org.partiql.lang.errors.Problem
+import org.partiql.lang.errors.ProblemDetails
+import org.partiql.lang.errors.ProblemSeverity
+import org.partiql.lang.eval.physical.SetVariableFunc
+import org.partiql.lang.eval.physical.operators.RelationExpression
+import org.partiql.lang.eval.physical.operators.ScanRelationalOperatorFactory
+import org.partiql.lang.eval.physical.operators.ValueExpression
+import org.partiql.lang.eval.physical.sourceLocationMetaOrUnknown
+import org.partiql.lang.planner.PartiQLPlanner
+import org.partiql.lang.planner.PartiQLPlannerPass
+import org.partiql.lang.planner.PlannerEventCallback
+import org.partiql.lang.planner.PlanningProblemDetails
+import org.partiql.lang.planner.createFakeGlobalsResolver
+import org.partiql.lang.planner.transforms.DEFAULT_IMPL_NAME
+import org.partiql.lang.planner.transforms.PLAN_VERSION_NUMBER
+import org.partiql.lang.util.PartiQLExperimental
+
+@PartiQLExperimental
+class PartiQLCompilerPipelineSmokeTests {
+
+    private fun createPlannerPipelineForTest(
+        allowUndefinedVariables: Boolean,
+        plannerEventCallback: PlannerEventCallback?,
+        block: PartiQLCompilerPipeline.Builder.() -> Unit = { }
+    ) = PartiQLCompilerPipeline.build {
+        compiler
+            .ionSystem(ION)
+        planner.options(
+            PartiQLPlanner.Options(
+                allowedUndefinedVariables = allowUndefinedVariables,
+            )
+        ).callback {
+            plannerEventCallback?.invoke(it)
+        }.globalVariableResolver(createFakeGlobalsResolver("Customer" to "fake_uid_for_Customer"))
+        block()
+    }
+
+    @Test
+    fun `happy path`() {
+        var pecCallbacks = 0
+        val plannerEventCallback: PlannerEventCallback = { _ ->
+            pecCallbacks++
+        }
+
+        val pipeline = createPlannerPipelineForTest(allowUndefinedVariables = true, plannerEventCallback = plannerEventCallback)
+
+        // the constructed ASTs are tested separately, here we check the compile function does not throw any exception.
+        assertDoesNotThrow {
+            pipeline.compile("SELECT c.* FROM Customer AS c WHERE c.primaryKey = 42")
+        }
+
+        // pec should be called once for each pass in the planner:
+        // - normalize ast
+        // - ast -> logical
+        // - logical -> logical resolved
+        // - logical resolved -> default physical
+        assertEquals(4, pecCallbacks)
```

Comment:
This test result is impacted by the redesign. 
Before it was 6 callbacks, `parse, normalize ast, logical, logical resolved, default physical, compile`.
Now it was 4, because parse and compile are not the planner's (`plan()`) responsibility.

Reply:
Ack. This was intentional. We didn't want the planner to also parse and normalize — just plan